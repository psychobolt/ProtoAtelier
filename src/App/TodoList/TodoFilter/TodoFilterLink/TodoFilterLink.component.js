// @flow
import React from 'react';

const TodoFilterLink = (
  { active, goto, children }: {active?: boolean, goto: Function, children: any },
) => (
  <x-tab
    onClick={goto}
    selected={active || null}
  >
    <x-label>{children}</x-label>
  </x-tab>
);

TodoFilterLink.defaultProps = {
  active: false,
};

export default TodoFilterLink;
