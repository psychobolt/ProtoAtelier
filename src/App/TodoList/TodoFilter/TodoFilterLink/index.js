export { default } from './TodoFilterLink.container';
