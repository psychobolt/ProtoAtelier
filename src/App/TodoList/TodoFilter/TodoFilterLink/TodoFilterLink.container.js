import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

import TodoFilterLink from './TodoFilterLink.component';
import Filters from '../TodoFilter.filters';

const mapParamToFilter = param => param || Filters.ALL;

export const ROOT = '/';

const mapStateToProps = (state, ownProps) => ({
  active: ownProps.filter === mapParamToFilter(ownProps.match.params.filter),
  goto: () => {
    ownProps.history.push(ownProps.filter === Filters.ALL ? ROOT : ownProps.filter);
  },
});

export default withRouter(connect(mapStateToProps)(TodoFilterLink));
