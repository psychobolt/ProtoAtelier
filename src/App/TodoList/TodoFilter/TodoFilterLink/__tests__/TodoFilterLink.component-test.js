import React from 'react';
import { shallow } from 'enzyme';

import TodoFilterLink from '../TodoFilterLink.component';
import Filters from '../../TodoFilter.filters';

describe('components <TodoFilterLink />', () => {
  it('TodoFilterLink should render correctly -- active link', () => {
    const props = {
      active: true,
      children: <span>Link</span>,
      filter: Filters.ALL,
    };
    const wrapper = shallow(<TodoFilterLink {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('TodoFilterLink should render correctly -- completed filter', () => {
    const props = {
      children: <span>Link</span>,
      filter: Filters.COMPLETED,
    };
    const wrapper = shallow(<TodoFilterLink {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('TodoFilterLink should render correctly -- all filter', () => {
    const props = {
      children: <span>Link</span>,
      filter: Filters.ALL,
    };
    const wrapper = shallow(<TodoFilterLink {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
