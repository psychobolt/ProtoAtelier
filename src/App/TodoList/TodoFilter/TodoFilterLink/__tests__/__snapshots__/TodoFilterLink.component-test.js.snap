// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components <TodoFilterLink /> TodoFilterLink should render correctly -- active link 1`] = `
<x-tab
  selected={true}
>
  <x-label>
    <span>
      Link
    </span>
  </x-label>
</x-tab>
`;

exports[`components <TodoFilterLink /> TodoFilterLink should render correctly -- all filter 1`] = `
<x-tab
  selected={null}
>
  <x-label>
    <span>
      Link
    </span>
  </x-label>
</x-tab>
`;

exports[`components <TodoFilterLink /> TodoFilterLink should render correctly -- completed filter 1`] = `
<x-tab
  selected={null}
>
  <x-label>
    <span>
      Link
    </span>
  </x-label>
</x-tab>
`;
